### DISCUSSION TITLE
Soft off support for 1.1 technikable PCB

### DISCUSSION THREAD

**Discussion by GrantCuster** [manna-harbour/miryoku#315] [Q&A] [answered]
I just flashed a new technikable pcb with Miryoku and everything is working great except the 'soff off' which I gather is a pretty new feature.

I found the technikable revision https://github.com/petejohanson/technikable-zmk-config/commit/60b1d2efb910018446ecd7e6e1f0bd7ea15486c8#diff-870dfcd95b0ae6a9a44d71c7a445eef66af4bc8e9c6d9a7e788957ba94d676fe and also this discussion of the soft off feature in relation to miryoku here https://github.com/manna-harbour/miryoku/discussions/284. But I'm not totally sure what my process should be for enabling soft off for the technikable 1.1. I did try just adding CONFIG_ZMK_PM_SOFT_OFF=y to KCONFIG but that was not enough. Not sure what else to apply from that overlay.

Any tips? Thanks!

**Comment by manna-harbour** [comment]
It looks like `technikable@1.1` has hardware support for soft off. I'll add support for that board revision, but if you'd like to do it you could try adding an outboard for `boards/technikable@1.1`. See <https://github.com/manna-harbour/miryoku_zmk#outboards>.

**Comment by GrantCuster** [comment]
Thanks for taking a look! If I have time to tinker I'll take a look at adding the outboard, but I'll probably wait for it to be added.

**Comment by GrantCuster** [comment]
I ended up tinkering -- it's a little weird (I think) because 1.1 is in the same folder as the original technikable, so I couldn't figure out a way to specify which revision in the outboard config file.

I ended up forking zmk-config and setting the default revision to 1.1
https://github.com/GrantCuster/technikable-zmk-config/commit/6129f77f5345870fbe860fd4a6be0361855e5868

Then pointing the technikable outboard config in my fork of miryoku zmk to that config fork. Soft off and everything else working great for me now.

**Reply by manna-harbour** [reply]
For revisions, you build with the board name including the revision, e.g. `technikable@1.1`. Outboard file names need to match the board or shield name being built, e.g. `boards/technikable@1.1`. In this case both `boards/technikable` and `boards/technikable@1.1` will have the same contents, as the same keyboard definition supports multiple board revisions.

The workflow could be updated to look for an outboard file without the board revision, if none is found with the revision, then there would only need to be one file in this case.

**Reply by manna-harbour** [reply]
The workflow now strips the revision before matching the outboard.

**Comment by manna-harbour** [comment, marked as answer]
I've added the outboard. Build with board `technikable@1.1`. Here's a sample build run: https://github.com/manna-harbour/miryoku_zmk/actions/runs/8903167272.

@petejohanson I've also updated the original `technikable` outboard to use the module.
